scipy/interpolate/fitpack2.py

```diff
@@ -900,6 +900,11 @@ def __call__(self, x, y, dx=0, dy=0, grid=True):
             if x.size == 0 or y.size == 0:
                 return np.zeros((x.size, y.size), dtype=self.tck[2].dtype)
 
+            if (x.size >= 2) and (not np.all(np.diff(x) >= 0.0)):
+                raise ValueError("x must be strictly increasing when `grid` is True")
+            if (y.size >= 2) and (not np.all(np.diff(y) >= 0.0)):
+                raise ValueError("y must be strictly increasing when `grid` is True")
```

Comment:
Probably fairly obvious, but there's no test for the second exception here (for `y`). I was just browsing through the codecov patch diff, but I see those are extremely similar errors so may not matter much.

I'm not an expert on this code at all, so fairly superficial observation.

Reply:
Thank you for reviewing. I fixed and cleaned up the tests.